pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -168,4 +168,7 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubeschedulerconfigv1alpha1.Kub
 		enableContentionProfiling := true
 		obj.EnableContentionProfiling = &enableContentionProfiling
 	}
+	if obj.FeatureGates == nil {
```

Comment:
why is this necessary? can we be resilient to nil instead?

Reply:
IMO, if we don't test here, we'll have to do the same thing elsewhere

Reply:
It seems that this object is only used in `utilfeature.DefaultMutableFeatureGate.SetFromMap`, which uses a `for range`. So it might just work with nil. If it does, we can get rid of all those empty maps in options_test